contribs/ev/src/main/java/org/matsim/contrib/ev/routing/EvNetworkRoutingModule.java

```diff
@@ -104,11 +104,20 @@ public List<? extends PlanElement> calcRoute(RoutingRequest request) {
 		final Person person = request.getPerson();
 
 		List<? extends PlanElement> basicRoute = delegate.calcRoute(request);
+		List<Leg> nonWalkLegs = basicRoute.stream()
+			.filter(Leg.class::isInstance) // only Leg elements
```

Comment:
Just FYI: THere is also `TripStructureUtils.getLegs(final List<? extends PlanElement> planElements )`